app/utils/tools.py

```diff
@@ -71,26 +71,20 @@ def fuzzy_search_user(
     `query` will be compared against `users` name, firstname and nickname.
     The size of the answer can be limited using `limit` parameter.
 
-    Use RapidFuzz library
+    Use Jellyfish library
     """
 
-    # We can give a dictionary of {object: string used for the comparison} to the extract function
-    # https://maxbachmann.github.io/RapidFuzz/Usage/process.html#extract
-
     # TODO: we may want to cache this object. Its generation may take some time if there is a big user base
-    choices = []
+    names = [f"{user.firstname} {user.name} {user.nickname}" for user in users]
 
-    for user in users:
-        choices.append(f"{user.firstname} {user.name} {user.nickname}")
+    scored = [
+        (user, jaro_winkler_similarity(query, choice))
+        for user, choice in zip(users, names, strict=True)
+    ]
 
-    results: list[tuple[str, int | float, int]] = process.extract(
-        query,
-        choices,
-        limit=limit,
-    )
+    scored.sort(key=lambda x: x[1], reverse=True)
 
-    # results has the format : (string used for the comparison, similarity score, index of the object in the choices collection)
-    return [users[res[2]] for res in results]
+    return [user for user, _ in scored[:limit]]
```

Comment:
I'm not sure what could be a good way to sort only the first `limit` elements. Currently, we sort 1000 users to return only the first 10

Reply:
The new system should be better